From a0943bd277ed9b5ecaf1c7c1ace7590b1b6da37d Mon Sep 17 00:00:00 2001
From: Shilin Victor <chrono.monochrome@gmail.com>
Date: Mon, 20 Aug 2018 22:57:43 +0300
Subject: [PATCH] check whether the ActivityClientRecord of token is null when handleStopActivity.

This is to fix the nullpointer problem happen in the application process because
of activity of the token is not found.

Test: monkey test 10h with no reference problem happen.

diff --git a/core/java/android/app/ActivityThread.java b/core/java/android/app/ActivityThread.java
index d242ba1..40c5df0 100644
--- a/core/java/android/app/ActivityThread.java
+++ b/core/java/android/app/ActivityThread.java
@@ -4083,6 +4083,10 @@
 
     private void handleStopActivity(IBinder token, boolean show, int configChanges, int seq) {
         ActivityClientRecord r = mActivities.get(token);
+        if (r == null) {
+            Log.w(TAG, "handleStopActivity: no activity for token:" + token);
+            return;
+        }
         if (!checkAndUpdateLifecycleSeq(seq, r, "stopActivity")) {
             return;
         }
